pkg/integrations/integrations.go

```diff
@@ -0,0 +1,67 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/k8sgpt-ai/k8sgpt-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	backstageLabelKey = "backstage.io/kubernetes-id"
```

Comment:
Can't this functionality be completely generic? Why not put a `Labels` field on the Result CRD to support adding arbitrary labels to the resource associated with the Result?

Reply:
It's the other way around, we search if resources in our results CRs which are generated from the operator include backstage label, if they do then we assign the same label to results CR of the respective resource.

I also don't expect users changing results CR which is fully managed from the operator.

Reply:
Thanks for the clarification. I still think it could be made fully generic. What if we copy all labels of the source resource onto the Result CR in all cases? That functionality could also be controlled by a flag for users to opt-in.

Just trying to consider how this type of ask can be handled in a maximally maintainable way.

Reply:
I think we need to tread lightly with labels, plenty of use cases that labels of resources correspond to ownership (helm labels) or external funcionality (kyverno, etc..) inserting all the resource labels to its Result CR may have undesirable effect for users. We can discuss this more thoroughly in slack or at the next meeting ?

Reply:
What about adding a `CopyLabels` field to the Result CR that the controller refers to each time it registers a result? The controller would iterate over all labels of the resource associated with the Result and copy any labels that are found in `CopyLabels`. Definitely less efficient, but more extensible.

Happy to discuss further in Slack.

Reply:
My take for now is to merge the backstage support, small PR and focus our efforts on feature parity with k8sgpt and if we get more use cases for label support we build a more generic functionality when we have more context.